### ISSUE TITLE
Recognition of greetings 

### ISSUE THREAD

**Issue by albiony** [swapagarwal/JARVIS-on-Messenger#382]
The greetings JARVIS gives themselves ([here](https://github.com/swapagarwal/JARVIS-on-Messenger/blob/master/modules/src/hello.py#L8-L14)) are fine but I found a simple 'Hi!' throws an error message. 
I had this issue with most greetings apart from a simple 'Hello' and 'Hi' (no punctuation). This meant I ended up meeting a lot of 'I'm sorry;...' messages very early on.

I'm very new to open source but would love to help to expand upon the types of greetings JARVIS recognises, though I'm not sure how.

Love the whole JARVIS idea by the way!
![jarvis issue 1](https://user-images.githubusercontent.com/36374886/36105190-eb991448-100b-11e8-919b-8f80cb8ca12f.png)

**Comment by sync350** [comment]
Could we create a dictionary or list of common greetings? And is there a way to ignore punctuation? 
I'd be willing to take a stab at this one.

**Comment by albiony** [comment]
Maybe something like this could work? The 'for' bit could be improved but it shows how you could strip punctuation out of the message:
![issue 1_feedback](https://user-images.githubusercontent.com/36374886/36144887-72db8d82-10a7-11e8-8c00-267ff4ff86d2.png)

**Comment by sync350** [comment]
Yeah, I think I could make that work. I'll go ahead and get working on this one.

**Comment by albiony** [comment]
Thanks! Would you mind posting a link to the changes? Really curious to see how this one's resolved.

**Comment by TommyMGun** [comment]
could probably use regex + dictionary

**Comment by albiony** [comment]
Maybe something like this? Gets rid of the punctuation stripping function and seems to work well with the existing list in the hello.py module.
![issue 1_feedback1](https://user-images.githubusercontent.com/36374886/36341424-fdb3c9a6-13e5-11e8-8fef-0e86866b1806.png)

**Comment by sync350** [comment]
So I'm taking a look through the code, and it's been a while since I've contributed to this project. I can't seem to find where it actually parses the input to trigger the greetings in hello.py?

**Comment by albiony** [comment]
So as I understand it:

The [hello.py](https://github.com/swapagarwal/JARVIS-on-Messenger/blob/master/modules/src/hello.py) module defines the input/output process. Then in the [main bot](https://github.com/swapagarwal/JARVIS-on-Messenger/blob/master/jarvis.py) this is converted into a 'process()' function (line 22). Later on line 56 there's a request but I can't really make sense of it.

But I think I'm in the same boat as you at the moment. I'll keep trying and let you know if I come up with anything. Hopefully someone around can lend a hand?

**Comment by addyvm22** [comment]
hello @albiony @wskinner74 the bot just sends the text entered to wit.ai and wit then returns a processed output giving the possible intent of the user and the confidence of the answer. If the confidence>50% only then is the intent considered as good. I checked these values for the specific string 'hi!' it returned intent='hello' but the confidence is just 0.352.  Thus the intent is not accepted. That is really weird from wit as a string like 'hi there!!!!!!!!' returns an intent 'hello' and a confidence of 0.935.

here's code to check what wit returns for various messages entered..

```
"""
Created on Tue Mar 20 11:26:17 2018

@author: Adwait
"""
import config
import requests

token=config.WIT_AI_ACCESS_TOKEN
while(True):
    message=input("input your message here  ")
    if message=='qt':
        break    
    r = requests.get('https://api.wit.ai/message?v=20160420&q=' + message , headers={
                'Authorization': 'Bearer %s' % token} )
    witreply=r.json()
    intent = witreply['outcomes'][0]['intent']
    print(intent)
    print(witreply)
```

**Comment by addyvm22** [comment]
I guess we should just add some of these exceptions to the wit notepad file on our pc--- JARVIS-on-Messenger-master\local\wit
